### DISCUSSION TITLE
Modification – Conversational UI copy sizes on mobile

### DISCUSSION THREAD

**Discussion by michelleallendavey** [Legal-and-General/canopy#786] [Canopy requests]
Hi all, the conversational copy on the PSP hero doesn't change much in size between mobile and desktop and is quite big when there's more than a few words on mobile: https://legal-and-general.github.io/canopy/?path=/story/components-hero--conversational-ui

Would like to explore if we can easily display a different font size on SM / MD / LG > XL breakpoints, or if we'd need to revisit the font-sizes generally?
![Hero_size6_SM@1x](https://user-images.githubusercontent.com/70210216/161726449-cbe45a4a-8b27-412f-bb3e-c940bfe771a1.jpg)

**Comment by elenagarrone** [comment]
@michelleallendavey from a development point of view we could change the style for the `lg-hero-card-title` (the big text) to change on different breakpoints 👍 
I guess the only thing to consider is having to change also the other components that might be displayed together with the `lg-hero-card-title` e.g. https://legal-and-general.github.io/canopy/?path=/story/components-hero--product-details so that everything is resizing appropriately.

**Reply by michelleallendavey** [reply]
Ah thanks @elenagarrone, good to see what else would be affected... will have a tinker with them and take to the designers in Thursday's sync. Watch this space 😎

**Reply by owensgit** [reply]
One thing we could consider is reducing just a little for now until you've had a chance to assess everything holistically? Maybe from `lg-font-size-6` to `lg-font-size-5` or maybe `lg-font-size-4`. I'd be happy to do this on `localhost` on a call so you can see it immedialtly. This would be a very small and quick change that could bring some immeidate benefit.

**Reply by alexcanning** [reply]
@michelleallendavey there have been some previous concepts which have aimed to reduce the type size of the headings and make that component work more effectively

**Reply by michelleallendavey** [reply]
Yeah I think I worked on something way back at the end of 2020/early 2021 – has anything happened more recently @alexcanning? Thanks @owensgit, that would be good – will give you a shout on Slack (probably tomorrow if you're around)

**Reply by alexcanning** [reply]
No, they were just concepts - feel free to pick it up!

**Reply by michelleallendavey** [reply]
I reviewed this with a couple of the other creative designers yesterday – with the conversational PSP hero type style being linked to the product hero title, there's not really a happy medium where the conversational UI is small enough while the product hero title is big enough... (it starts to become similar to, or smaller than, the big data point on the product hero.)

We wondered about the feasibility of two options:

1.  (Preferred option) Splitting out the conversational copy so that "Good morning, [name]" or equivalent uses `lg-hero-card-title`, and then any further copy ("your life and home are covered" "your investments are worth X") are in a smaller text size – would affect all products, so not a small piece of work I assume?
2. Unlinking the conversational copy on the PSP from `lg-hero-card-title` and creating a different style for that one, so that we can make it smaller independently of the product hero title.

**Reply by michelleallendavey** [reply]
@owensgit @elenagarrone @aaronBery Hello! Would appreciate thoughts on the suggestions above if you get a sec, please.

**Reply by elenagarrone** [reply]
Really sorry @michelleallendavey - for some reason I don't get notifications when someone replies to discussions 😞   - am I the only one or is this a bug with the discussions? 🤔 

I guess we could update the `lg-hero-card-title` so that it takes an input of `size="sm | lg"` (or similar) which sets the font size. The default value could be set to be the largest one so that existing projects wouldn't have a breaking change.

So the structure could look like:
```
<lg-hero-card>
  <lg-hero-card-header>
    <lg-hero-card-title headingLevel="2">
      Good morning, Ben.
   </lg-hero-card-title>
    <lg-hero-card-title headingLevel="2" size="sm">
      We've got your retirement covered.
   </lg-hero-card-title>
```
I think it's fine to have two of the same heading levels close to each other but might be worth checking with DIG.

**Reply by michelleallendavey** [reply]
I think that makes a lot of sense – thanks @elenagarrone!

**Comment by st3fun1** [comment]
There is even a bigger issue when all the letters are capitalised:
![image](https://user-images.githubusercontent.com/12414812/170461470-aaf01796-f933-4305-89cd-a97326d4888f.png)

Using **lg-font-size-5** helps:
![image](https://user-images.githubusercontent.com/12414812/170461770-d0707010-093f-41de-8bf7-5c39c73d4d8d.png)
